tasks/st0361_uploading_facts.rb

```diff
@@ -0,0 +1,13 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+require 'puppet'
+require 'facter'
+
+puppet_bin_dir = if Facter.value(:osfamily).eql? 'windows'
+                   'C:\\"Program Files"\\"Puppet Labs"\\Puppet\\bin\\'
+                 else
+                   '/opt/puppetlabs/bin/'
+                 end
+
+factsupload = `#{puppet_bin_dir}puppet facts upload`
+
+puts factsupload
```

Comment:
This should be structured output based on the exit status of the command.

Reply:
Yeah this is what i did after i spent an hour failing at doing a proper structured return :) but start as i mean to go on! ill give it another shot

Reply:
OK should be complete